controllers/SiteController.php

```diff
@@ -108,4 +112,59 @@ public function actionAbout()
     {
         return $this->render('about');
     }
+
+    /**
+     * @param null $token
+     * @return string
+     * @throws \yii\base\Exception
+     * @throws CannotSaveException
+     */
+    public function actionValidateAccount($token = null)
+    {
+        $user = $this->checkIfUserExist($token);
+
+        $form = new ValidateAccountForm();
+
+        if($form->load($_POST) && $form->validate()) {
+            $this->validateAccount($form, $user);
+        }
+
+        return $this->render('validate-account', [
+           'model' => $form,
+        ]);
+    }
+
+    /**
+     * @param $form
+     * @param $user
+     * @throws CannotSaveException
+     * @throws \yii\base\Exception
+     */
+    private function validateAccount($form, $user)
+    {
+        $user->password_hash = Yii::$app->security->generatePasswordHash($form->password);
+        $user->is_validated = true;
+        $user->token = null;
+
+        if (!$user->save()) {
+            throw new CannotSaveException($user);
+        }
+
+        Yii::$app->user->login($user);
+        $_SESSION["userPasswordHash"] = $user->password_hash;
+        return $this->redirect("/");
+    }
+
+    /**
+     * @param $token
+     * @return User|null
+     */
+    private function checkIfUserExist($token)
```

Comment:
You should name this method checkIfUserMatchToToken cause it's clearly what this method does.

Reply:
I fully agree

Reply:
True